src/backends/beamcoder.ts

```diff
@@ -296,17 +300,16 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
         // Example: when we got a frame a 0 and request a frame at t = 30s just after, we don't want to start reading all packets
         // until 30s.
         const RE_SEEK_THRESHOLD = 3; // 3 seconds - typically we have keyframes at shorter intervals
-        const timeDifference = this.ptsToTime(Math.abs(targetPTS - (this.#packet?.pts || 0)));
-
-        VERBOSE && console.log(`timeDifference: ${timeDifference}, targetPTS: ${targetPTS}, last packet pts: ${this.#packet?.pts}`);
+        const hasFrameWithinThreshold = this.#filteredFramesPacket.flat().some(frame => {
```

Comment:
Maybe we should have some logic here:

```pseudocode
if targetPTS > stream's duration
    don't check RE_SEEK_THRESHOLD,
    instead, seek to stream's duration.
```

Reply:
Yeah, I was also thinking about that (see my todo). However, I want to do that in another PR. This one already addresses 3 different issues.

Reply:
I would expect that this allows to avoid recursively seeking? But we can merge like this and maybe we'll be able to remove the recursivity later. or maybe not. Let's see!

Reply:
I don't think we'll be able to remove it. See https://github.com/Lumen5/framefusion/blob/6741770ac92de0cb770318480caad77217e993e6/test/framefusion.test.ts#L248. We request frame `12.9` while the video's duration is `12.92` so `targetPTS is < stream's duration` in this case.

Reply:
🤔 🤔 🤔